Mathlib/Topology/EMetricSpace/Lipschitz.lean

```diff
@@ -362,19 +376,28 @@ protected lemma comp  {f : β → γ} {g : α → β}
     ((mapsTo_preimage g u).mono_left inter_subset_right)
 
 /-- If `f` and `g` are locally Lipschitz, so is the induced map `f × g` to the product type. -/
-protected lemma prod {f : α → β} (hf : LocallyLipschitz f) {g : α → γ} (hg : LocallyLipschitz g) :
+protected lemma prodMk {f : α → β} (hf : LocallyLipschitz f) {g : α → γ} (hg : LocallyLipschitz g) :
     LocallyLipschitz fun x => (f x, g x) := by
   intro x
   rcases hf x with ⟨Kf, t₁, h₁t, hfL⟩
   rcases hg x with ⟨Kg, t₂, h₂t, hgL⟩
   refine ⟨max Kf Kg, t₁ ∩ t₂, Filter.inter_mem h₁t h₂t, ?_⟩
-  exact (hfL.mono inter_subset_left).prod (hgL.mono inter_subset_right)
+  exact (hfL.mono inter_subset_left).prodMk (hgL.mono inter_subset_right)
+
+@[deprecated (since := "2025-03-10")]
+protected alias prod := LocallyLipschitz.prodMk
+
+protected theorem prodMk_left (a : α) : LocallyLipschitz (Prod.mk a : β → α × β) :=
+  (LipschitzWith.prodMk_left a).locallyLipschitz
+
+@[deprecated (since := "2025-03-10")]
+protected alias prod_mk_left := LocallyLipschitz.prodMk_left
 
-protected theorem prod_mk_left (a : α) : LocallyLipschitz (Prod.mk a : β → α × β) :=
-  (LipschitzWith.prod_mk_left a).locallyLipschitz
+protected theorem prodMk_right (b : β) : LocallyLipschitz (fun a : α => (a, b)) :=
```

Comment:
Note that these lemmas are named different from the corresponding injectivity lemmas: #naming says
> An injectivity lemma that uses "left" or "right" should refer to the argument that "changes". For example, a lemma with the statement a - b = a - c ↔ b = c could be called sub_right_inj.
Such a statement is not made here - so changing this would be good in a follow-up. Do you think we should change this, though? (Happy to start a discussion on zulip.)

Reply:
> Note that these lemmas are named different from the corresponding injectivity lemmas: #naming says
> > An injectivity lemma that uses "left" or "right" should refer to the argument that "changes". For example, a lemma with the statement a - b = a - c ↔ b = c could be called sub_right_inj.
> Such a statement is not made here - so changing this would be good in a follow-up. Do you think we should change this, though? (Happy to start a discussion on zulip.)

Do you want me to change the names as a part of this PR, or is it a suggestion for the future?

Reply:
Definitely a comment for the future: let's not have this PR rot.

It's a question to you: do you think it's worth aligning these rules, and if so, would you like to start a zulip topic (or shall I)?